test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -728,6 +785,96 @@ public void CreateDisplayMetadata_IsFlagsEnum_ReflectsModelType(Type type, bool
         }
 
         [Fact]
+        public void CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_IStringLocalizer()
+        {
+            // Arrange
+            var provider = CreateIStringLocalizerProvider();
+
+            var key = ModelMetadataIdentity.ForType(typeof(EnumWithLocalizedDisplayNames));
+            var attributes = new object[0];
+
+            var context = new DisplayMetadataProviderContext(key, new ModelAttributes(attributes));
+            provider.CreateDisplayMetadata(context);
+
+            // Act & Assert
+            using (new CultureReplacer("fr-FR", "fr-FR"))
+            {
+                Assert.All(
+                    context.DisplayMetadata.EnumGroupedDisplayNamesAndValues
+                        .Where(kvp => kvp.Value == "2"),
+                    kvp => Assert.Equal("Loc_Two_Name fr-FR", kvp.Key.Name));
```

Comment:
This tests identical functionality to `CreateDisplayMetadata_EnumGroupedDisplayNamesAndValues_NameWithIStringLocalizer` anyway, so I'm just going to delete it unless I'm misunderstanding you and your real problem is with what this test is covering.